File: easy/python/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        hash = {}
        for char in s:
            if hash.get(char) is None:
                hash[char] = 1
            else:
                hash[char] += 1

        for char in t:
            if hash.get(char) and hash[char] >= 1:
                hash[char] -= 1
            else:
                return False

        return True

File: easy/ruby/valid_anagram.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_anagram(s, t)
  return false if s.length != t.length

  char_count = Hash.new(0)

  s.each_char { |char| char_count[char] += 1 }

  t.each_char do |char|
    return false unless char_count.key?(char) && char_count[char].positive?

    char_count[char] -= 1
  end

  true
end
